ext/js/language/ja/shinjitai-converter.js

```diff
@@ -0,0 +1,515 @@
+    ['祕', '秘'],
+    ['碑', '碑'],
+    ['濱', '浜'],
+    ['賓', '賓'],
+    ['頻', '頻'],
+    ['敏', '敏'],
+    ['甁', '瓶'],
+    ['侮', '侮'],
+    ['福', '福'],
+    ['拂', '払'],
+    ['佛', '仏'],
+    ['倂', '併'],
+    ['竝', '並'],
+    ['塀', '塀'],
+    ['餠', '餅'],
+    ['邊', '辺'],
+    ['變', '変'],
+    ['勉', '勉'],
+    ['步', '歩'],
+    ['舖', '舗'],
+    ['寶', '宝'],
+    ['豐', '豊'],
+    ['襃', '褒'],
+    ['墨', '墨'],
+    ['沒', '没'],
+    ['飜', '翻'],
+    ['每', '毎'],
+    ['萬', '万'],
+    ['滿', '満'],
+    ['免', '免'],
+    ['麵', '麺'],
+    ['默', '黙'],
+    ['彌', '弥'],
+    ['譯', '訳'],
+    ['藥', '薬'],
+    ['與', '与'],
+    ['豫', '予'],
+    ['餘', '余'],
+    ['譽', '誉'],
+    ['搖', '揺'],
+    ['樣', '様'],
+    ['謠', '謡'],
+    ['來', '来'],
+    ['賴', '頼'],
+    ['亂', '乱'],
+    ['覽', '覧'],
+    ['欄', '欄'],
+    ['龍', '竜'],
+    ['隆', '隆'],
+    ['旅', '旅'],
+    ['虜', '虜'],
+    ['兩', '両'],
+    ['獵', '猟'],
+    ['綠', '緑'],
+    ['淚', '涙'],
+    ['壘', '塁'],
+    ['類', '類'],
+    ['禮', '礼'],
+    ['勵', '励'],
+    ['戾', '戻'],
+    ['靈', '霊'],
+    ['隸', '隷'],
+    ['齡', '齢'],
+    ['曆', '暦'],
+    ['歷', '歴'],
+    ['戀', '恋'],
+    ['連', '連'],
+    ['廉', '廉'],
+    ['練', '練'],
+    ['鍊', '錬'],
+    ['爐', '炉'],
+    ['勞', '労'],
+    ['郞', '郎'],
+    ['朗', '朗'],
+    ['廊', '廊'],
+    ['樓', '楼'],
+    ['籠', '篭'],
+    ['錄', '録'],
+    ['灣', '湾'],
+
+    // 人名用漢字
+    ['堯', '尭'],
+    ['巖', '巌'],
+    ['摑', '掴'],
+    ['彥', '彦'],
+    ['檜', '桧'],
+    ['槇', '槙'],
+    ['渚', '渚'],
+    ['猪', '猪'],
+    ['琢', '琢'],
+    ['瑤', '瑶'],
+    ['禰', '祢'],
+    ['祐', '祐'],
+    ['禱', '祷'],
+    ['祿', '禄'],
+    ['禎', '禎'],
+    ['穰', '穣'],
+    ['簞', '箪'],
+    ['聰', '聡'],
+    ['蓮', '蓮'],
+    ['蘭', '蘭'],
+    ['遙', '遥'],
+    ['遼', '遼'],
+    ['靖', '靖'],
+
+    // 表外字 (擴張新字體)
+    ['蘒', '蘒'],
+    ['啞', '唖'],
+    ['噓', '嘘'],
+    ['穎', '頴'],
+    ['鷗', '鴎'],
+    ['軀', '躯'],
+    ['鶯', '鴬'],
+    ['攪', '撹'],
+    ['麴', '麹'],
+    ['鹼', '鹸'],
+    ['嚙', '噛'],
+    ['繡', '繍'],
+    ['蔣', '蒋'],
+    ['醬', '醤'],
+    ['搔', '掻'],
+    ['屛', '屏'],
+    ['幷', '并'],
+    ['濾', '沪'],
+    ['蘆', '芦'],
+    ['蠟', '蝋'],
+    ['彎', '弯'],
+    ['焰', '焔'],
+    ['礦', '砿'],
+    ['讚', '讃'],
+    ['顚', '顛'],
+    ['巓', '巔'],
+    ['醱', '醗'],
+    ['潑', '溌'],
+    ['輛', '輌'],
+    ['繫', '繋'],
+    ['瀆', '涜'],
+    ['儘', '侭'],
+    ['藪', '薮'],
+    ['蠅', '蝿'],
+    ['嬀', '媯'],
+    ['驒', '騨'],
+
+    // variants
+
+    // 常用漢字
+    ['鬥', '闘'],
+    ['鬪', '闘'],
+    ['鬬', '闘'],
+
+    // 人名用漢字
+    ['亙', '亘'],
+    ['凜', '凛'],
+    ['晄', '晃'],
+    ['晉', '晋'],
+    ['萠', '萌'],
+]);
+
+let regexString = '';
+for (const [kyuji] of shinjitaiMap) {
+    regexString += kyuji;
+}
+const regex = new RegExp('[' + regexString + ']', 'g');
+
+/**
+ * @param {string} str
+ * @returns {string}
+ */
+export function convertShinjitai(str) {
+    return str.replace(regex, (c) => {
+        const shinji = shinjitaiMap.get(c);
+        if (typeof shinji !== 'undefined') { return shinji; }
+        return c;
```

Comment:
A bit cleaner to do this for undefined checking.
```suggestion
        return shinjitaiMap.get(c) ?? c;
```